### PR TITLE
Added HUD, coin icon

### PR DESCRIPTION
**Pull request by Dwooll94**
Created a statically accessed HUD and added a coin icon. All numeric fields are fixed width and will trim off the front of the number if too large for HUD.

### CODE REVIEW

File: source/HeadsUpDisplay.hx [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+	/**
+	 * 
+	 * @param	newTime The time you want displayed on the HUD. Displays last 3 digits of number.
+	 */
+	private function handleTimeUpdate(newTime:Int):Void{
+		//trim number down to last 3 digits if over 3 digits
+		var parsedTime:String = Std.string(newTime);
+		var timeString:String = "000";
+		var parsedLength:Int = parsedTime.length;
+		timeString = timeString.substr(0, -parsedLength) + parsedTime;
```
Comment (KeithCissell) [inline source/HeadsUpDisplay.hx]: I noticed that in this function and in 'handleCoinsUpdate' you have a line to set a their string values, but in the if statement following that you either set that string to that same value again or another format. Seems redundant, but there may have been a reason you needed to do it this way.

**Review by KeithCissell — COMMENTED** [review]
The code is really clean and the output of the HUD looks really good, especially with the shadow styling. One thing you might look into next is how to make sure the HUD displays are above the map so the two don't overlap. I left a comment about possible redundant code and the update function. Another thing you might add would be a variable for a playClock. Then in the update function for the HUD, add logic that increments the timeClock every second and passes the new time to 'handleTimeUpdate'.

File: source/HeadsUpDisplay.hx [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+	/**
+	 * 
+	 * @param	newCoinCount the coin count you want displayed on the HUD. Displays last 2 digits of number.
+	 */
+	private function handleCoinsUpdate(newCoinCount:Int){
+		//trim number down to last 2 digits if over 2 digits
+		var parsedCoins:String = Std.string(newCoinCount);
+		var coinString:String = "00";
+		var parsedLength:Int = parsedCoins.length;
+		coinString = coinString.substr(0, -parsedLength) + parsedCoins;
```
Comment (KeithCissell) [inline source/HeadsUpDisplay.hx]: Other possible redundancy

File: source/HeadsUpDisplay.hx [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+		//UPDATES
+		scoreLabel.update(elapsed);
+		timeLabel.update(elapsed);
+		coinIcon.update(elapsed);
+		coinCross.update(elapsed);
+		score.update(elapsed);
+		time.update(elapsed); 
+		coins.update(elapsed);
+		
+
```
Comment (KeithCissell) [inline source/HeadsUpDisplay.hx]: You need to call 'super.update(elapsed);' at the very end of this update function

File: source/HeadsUpDisplay.hx [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+		score.text = scoreString;
+	}
+	/**
+	 * 
+	 * @param	elapsed
+	 */
+	
+	public override function update(elapsed:Float):Void
+	{
+		//UPDATES
```
Comment (KeithCissell) [inline source/HeadsUpDisplay.hx:126]: I don't think each of these items need below need to have update called on them. You would just put logic here that updated them using the functions you created above. For example, you would check the time elapsed and if it had been a full second call 'handelTimeUpdate()' with the new time.

**Review by SamBumgardner — CHANGES_REQUESTED** [review]
The documentation looks pretty good, but there is a change I'd like to have before it gets merged in.

At the top of every function's documentation block, put a short description of what the function is supposed to accomplish. See the `Player.hx` file for examples!

**Review by SamBumgardner — APPROVED** [review]

**Pull request merged**